samples/wifi/provisioning/implementation_details.rst

```diff
@@ -0,0 +1,195 @@
+Data Out                   14387803-130c-49e7-b877-2881c89cb258
+========================== ====================================
+
+The characteristic requirements of the Wi-Fi Provisioning Service are shown in the following table.
+
++-----------------+-------------+-------------+-------------+-------------+
+| Characteristic  | Requirement | Mandatory   | Optional    | Security    |
+| Name            |             | Properties  | Properties  | Permissions |
++=================+=============+=============+=============+=============+
+| Information     | Mandatory   | Read        |             | No Security |
+|                 |             |             |             | required    |
++-----------------+-------------+-------------+-------------+-------------+
+| Operation       | Mandatory   | Indicate,   |             | Encryption  |
+| Control         |             | Write       |             | Required    |
+| Point           |             |             |             |             |
++-----------------+-------------+-------------+-------------+-------------+
+| Operation       | Mandatory   | Read, Write |             | Encryption  |
+| Control         |             |             |             | Required    |
+| Point           |             |             |             |             |
+| - Client        |             |             |             |             |
+| Characteristic  |             |             |             |             |
+| Configuration   |             |             |             |             |
+| descriptor      |             |             |             |             |
++-----------------+-------------+-------------+-------------+-------------+
+| Data Out        | Mandatory   | Notify      |             | Encryption  |
+|                 |             |             |             | Required    |
++-----------------+-------------+-------------+-------------+-------------+
+| Data Out        | Mandatory   | Read, Write |             | Encryption  |
+| - Client        |             |             |             | Required    |
+| Characteristic  |             |             |             |             |
+| Configuration   |             |             |             |             |
+| descriptor      |             |             |             |             |
++-----------------+-------------+-------------+-------------+-------------+
+
+The purpose of each characteristic is as follows:
+
+* ``Information``: for client to get info message from server.
+* ``Operation Control Point``: for client to send request message to server, and server to send response message to client.
+* ``Data Out``: for server to send result message to the client.
+
+Advertisement
+-------------
+
+After powerup, the device checks if it is provisioned. If it is, it loads saved credentials and tries to connect to the specified access point.
+After 30 seconds, the device advertises over Bluetooth LE. The operation times out if the attempt fails.
```

Comment:
```suggestion

After 30 seconds, the device advertises over Bluetooth LE. The operation times out if the attempt fails.

```

Reply:
In current implementation, the device will try to apply saved wifi credential. Then, it no longer waits for the connection result, and starts Bluetooth advertisement immediately. So I think we can shorten this part.

> After powerup, the device checks if it is provisioned. If it is, it loads saved credentials and tries to connect to the specified access point.

>After 30 seconds, the device advertises over Bluetooth LE. The operation times out if the attempt fails.

>If the device is not provisioned, it advertises immediately.

How about this:

> After powerup, the device checks if it is provisioned. If it is, it loads saved credentials and tries to connect to the specified access point.

> Then, the device advertises over Bluetooth LE.